enterprise_subsidy/apps/subsidy/models.py

```diff
@@ -628,15 +629,25 @@ def _create_redemption(
         ledger_transaction.save()
 
         external_transaction_reference = None
-        try:
-            if self.geag_fulfillment_handler().can_fulfill(ledger_transaction):
+        if is_geag_fulfillment(ledger_transaction):
+            if not self.geag_fulfillment_handler().can_fulfill(ledger_transaction):
+                logger.error(
+                    f'Could not find variant id for transaction {ledger_transaction.uuid} '
+                    f'which can only be fulfilled via GEAG. Rolling back transaction.'
+                )
+                self.rollback_transaction(ledger_transaction)
+                raise IncompleteContentMetadataException(
+                    f'Missing variant_id needed for GEAG transaction {ledger_transaction}, not attempting fulfillment'
+                )
```

Comment:
Nit: I'd prefer we raise and just let the exception handler perform rollback so that we funnel all rollback scenarios to one line.

Reply:
ah, yup, that's cleaner